src/azure-cli/azure/cli/command_modules/resource/_help.py

```diff
@@ -1972,18 +1972,104 @@
 
 helps['tag create'] = """
 type: command
-short-summary: Create a tag in the subscription.
+short-summary: Create tags on a specific resource.
+long-summary: >
+    The az tag create command with an id creates or updates the entire set of tags on a resource, resource group or subscription.
+    This operation allows adding or replacing the entire set of tags on the specified resource, resource group or subscription.
+    The specified entity can have a maximum of 50 tags.
+parameters:
+  - name: --name -n
+    short-summary: The name of the tag to create.
+  - name: --subscription
+    short-summary: Name or ID of subscription. You can configure the default subscription using az account set -s NAME_OR_ID.
+  - name: --resource-id
+    short-summary: The resource identifier for the entity being tagged. A resource, a resource group or a subscription may be tagged.
+  - name: --tags
+    short-summary: The tags to be applied on the resource.
 examples:
   - name: Create a tag in the subscription.
     text: >
         az tag create --name MyTag
+  - name: Create or update the entire set of tags on a subscription.
+    text: >
+        az tag create --resource-id /subscriptions/{subId} --tags Dept=Finance Status=Normal
+  - name: Create or update the entire set of tags on a resource group.
+    text: >
+        az tag create --resource-id /subscriptions/{sub-id}/resourcegroups/{rg} --tags Dept=Finance Status=Normal
+  - name: Create or update the entire set of tags on a resource.
+    text: >
+        az tag create --resource-id /subscriptions/{sub-id}/resourcegroups/{rg}/providers/Microsoft.Compute/virtualMachines/{vmName} --tags Dept=Finance Status=Normal
 """
 
 helps['tag delete'] = """
 type: command
-short-summary: Delete a tag in the subscription.
+short-summary: Delete tags on a specific resource.
+long-summary:
+    The az tag delete command with an id deletes the entire set of tags on a resource, resource group or subscription.
+parameters:
+  - name: --name -n
+    short-summary: The name of the tag to be deleted.
+  - name: --resource-id
+    short-summary: The resource identifier for the entity being tagged. A resource, a resource group or a subscription may be tagged.
```

Comment:
I will recommend you define these common params in _params.py  with CLIArgumentType and reuse them command group level or each command.

Reply:
I have changed commonly used `name`, `value`, and 'resource-id' to `CLIArgumentType` in `_params.py`. Not sure if need to do any change here in `_help.py`.